### ISSUE TITLE
high FPKM called with few reads mapping to transcript

### ISSUE THREAD

**Issue by roryk** [gpertea/stringtie#12]
Thanks for making Stringtie-- I have been running it on some test data and it calls FPKM in places where there are almost no reads at all. 

I ran it like this:

```
stringtie -b stringtie/tx/tmpsZIpEI/Test1 -p 1 -G /v-data/bcbio-nextgen/tests/data/genomes/mm9/rnaseq/ref-transcripts.gtf -e /v-data/bcbio-nextgen/tests/test_automated_output/align/Test1/1_110907_ERP000591_tophat/Test1.bam
```

which generates this:

```
18	chrM	+	6942	7011	ENSMUST00000082404	1	70	ENSMUSG00000064353	mt-Td	8911.531250	2380952.250000
19	chrM	+	7013	7696	ENSMUST00000082405	1	684	ENSMUSG00000064354	mt-Co2	911.998718	243664.703125
20	chrM	+	7700	7764	ENSMUST00000082406	1	65	ENSMUSG00000064355	mt-Tk	9597.033203	2564102.500000
21	chrM	+	7766	7969	ENSMUST00000082407	1	204	ENSMUSG00000064356	mt-Atp8	3057.877197	816993.187500
22	chrM	+	7927	8607	ENSMUST00000082408	1	681	ENSMUSG00000064357	mt-Atp6	916.016357	244738.109375
23	chrM	+	8607	9390	ENSMUST00000082409	1	784	ENSMUSG00000064358	mt-Co3	795.672424	212585.046875
```

ENSMUST00000082404 has a very high FPKM but no reads map there.

IGV screenshot:

![igv_snapshot](https://cloud.githubusercontent.com/assets/414586/7274704/13daadfe-e8cd-11e4-9f41-eee076d45a71.png)

It doesn't look like it on IGV, but there are a couple of reads that overhang ENSMUST00000082404 by a couple of bases:

```
vagrant@vagrant-ubuntu-trusty-64:/v-data/bcbio-nextgen/tests$ samtools view test_automated_output/upload/Test1/Test1-ready.bam chrM:6942-7011
ERR032227.10543296	99	chrM	7008	50	76M	=	7233	301	CTTATATGGCCTACCCATTCCAACTTGGTCTACAAGACGCCACATCCCCTATTATAGAAGAGCTAATAAATTTCCA	HHHHHHHHHHHHHHHGHHHHFHHFHFHH?HFGGGEDFHEHEBGCBCFEF@EBECCB>>@;;A>??>B?>??=>?B=	MD:Z:76	RG:Z:1	XG:i:0	NH:i:1	NM:i:0	XM:i:0	XN:i:0	XO:i:0	AS:i:0	YT:Z:UU
ERR032227.10212620	99	chrM	7010	50	76M	=	7263	329	CCTATGGCCTACCCATTCCAACTTGGTCTACAAGACGCCACATCCCCTATTATAGAAGAGCTAATAAATTTCCATG	HHHHDHHHHHHHHHGHHGHDHECHEH;GFEHGCCFCCGCCGA?@CEE?<?B@?A>:>=;>??=>?;>><:?;@;;;	MD:Z:0T0A74	RG:Z:1	XG:i:0	NH:i:1	NM:i:2	XM:i:2	XN:i:0	XO:i:0	AS:i:-10	YT:Z:UU
ERR032227.10782111	161	chrM	7010	50	76M	=	7280	346	GTTATGGCCTACCCATTCCAACTTGGTCTACAAGACGCCACATCCCCTATTATAGAAGAGCTAATAAATTTCCATG	EEEEEEEEEEEEEEEEEEE?EAEEEE>BDEEEBE>=DB=?A>>??=BB<BA@B>A==>8>9A=:A=><;;=<=;<<	MD:Z:0T0A74	RG:Z:1	XG:i:0	NH:i:1	NM:i:2	XM:i:2	XN:i:0	XO:i:0	AS:i:-10	YT:Z:UU
ERR032227.10060210	99	chrM	7011	50	76M	=	7257	322	CTATGGCCTACCCATTCCAACTTGGTCTACAAGACGCCACATCCCCTATTATAGAAGAGCTAATAAATTTCCATGA	HGHHGFHHHHHHHHHHHHHHFGHHH=HHHHGDBGHFHGGFE@AECF=B@AAAC::D@>?C>???@AC><>=??<=<	MD:Z:0A75	RG:Z:1	XG:i:0	NH:i:1	NM:i:1	XM:i:1	XN:i:0	XO:i:0	AS:i:-5	YT:Z:UU
ERR032227.11020015	163	chrM	7011	50	76M	=	7191	256	CGATGGCCTACCCATTCCAACTTGGTCTACAAGACGCCACATCCCCTATTATAGAAGAGCTAATAAATTTCCATGA	EEEEEEEEEEEEEEEEEEEEBEEEE>EE8EEEE@EBEB@EB?ABAA?>>EA@??>=>:@<>?:;?<<@<===<<88	MD:Z:0A0T74	RG:Z:1	XG:i:0	NH:i:1	NM:i:2	XM:i:2	XN:i:0	XO:i:0	AS:i:-10	YT:Z:UU
ERR032227.11305478	99	chrM	7011	50	76M	=	7258	323	CGATGGCCTACCCATTCCAACTTGGTCTACAAGACGCCACATCCCCTATTATAGAAGAGCTAATAAATTTCCATGA	HHHHHHHHHHHHHHHHHHHHHHHHHCHHHHHHHGGGHFEGBCFBEAA?=C@?C@=>@>AE?=BAA<@;@B:?=@?5	MD:Z:0A0T74	RG:Z:1	XG:i:0	NH:i:1	NM:i:2	XM:i:2	XN:i:0	XO:i:0	AS:i:-10	YT:Z:UU
```

The FPKM is so high because this is a really tiny test dataset of about 50k reads.

Cufflinks and featureCounts both call this as having an expression level of zero, which seems correct to me. Is there something I could tweak to get similar results? I tried setting -g 1 and omitting -e but that didn't make any difference. I tried the brew installed 1.0.3 and also building from master, but had the same behavior.

Here are the Cufflinks FPKM calls for reference:

```
ENSMUST00000082404	0.0
ENSMUST00000082405	224658.0
ENSMUST00000082406	0.0
ENSMUST00000082407	24456.4
ENSMUST00000082408	234162.0
ENSMUST00000082409	0.0
```

I can provide all of the test data if that would be helpful. Thanks!

**Comment by gpertea** [comment]
Great bug report - thank you for the clear and detailed explanations. Indeed it seems that those few overlapping reads which are connecting the separate transcripts in the same bundle could also make StringTie report an inaccurate FPKM for that overhanging transcript.. 
Could you please upload that test data set on our ftp (ftp.ccb.jhu.edu) in the incoming/gpertea subdirectory? Thanks a lot!

**Comment by roryk** [comment]
Hi @gpertea,

Thanks for the reply, hope you are enjoying your weekend. The data is so tiny I just stuck it up on Dropbox. I included the FASTQ files, the Tophat alignments and the genome and GTF annotation I used, along with the Stringtie t_data.ctab output. The reference is just chrM so it is tiny. If I can include anything else that would help as well that isn't in there, just let me know. 

Here is the link: https://dl.dropboxusercontent.com/u/2822886/Test1.tar

**Comment by gpertea** [comment]
Got it - and thanks to your excellent report we identified the problem and it will be fixed in the next release.
Thank you!

**Comment by roryk** [comment]
Hi @gpertea,

Sweet, you guys rock. I'm down for testing a prerelease version on a more comprehensive dataset if you push your fixes back to github.

**Comment by schelhorn** [comment]
@gpertea: Hi, I'm assuming you guys use a private clone of this repository for buxfixes and only push to github for new releases. Could you us give an indication of when this bugfix will appear in an release that we can use? Thanks a bunch!

**Comment by gpertea** [comment]
Somewhat true - I am using the public github branches for working some bugfixes but we also have an internal SVN repo and I am only merging everything in the **master** branch here when it's ready for release.. 
Sorry guys for not responding to @roryk's request earlier, the CSHL conference delayed things a bit and I meant to signal a testing branch for this issue when I could confirm it was working properly on his example - but right now there are still a few details to work out for this FPKM miscall issue. I'll let you know when a testing branch is available here - though we may skip directly to a release version, since there were also a couple little bugfixes that I am eager to push out there.

**Comment by roryk** [comment]
Sweet, thanks for the followup @gpertea. Hope you had fun at CSHL.

**Comment by gpertea** [comment]
The devel branch here on github (which we are now testing for the imminent v1.0.4 release) should be ready for testing - and should have been fixed these (and other) issues.

**Comment by gpertea** [comment]
We just released v1.0.4  - @roryk or others who were waiting for it, please let us know if you encounter any problems with this new version..

**Comment by roryk** [comment]
Thanks @gpertea, good timing!
